MapboxCoreNavigation/RouteController.swift

```diff
@@ -81,6 +81,14 @@ public protocol RouteControllerDelegate: class {
      */
     @objc(routeController:didUpdateLocations:)
     optional func routeController(_ routeController: RouteController, didUpdate locations: [CLLocation])
+    
+    /**
+     Called when the route controller arrives at a waypoint.
+     - parameter waypoint: The waypoint that the controller has arrived at.
+     - parameter finalDestination: A boolean flag that signals that the waypoint is the final destination.
+    */
+    @objc(routeController:didArriveAtWaypoint:)
+    optional func routeController(_ routeController: RouteController, didArriveAt waypoint: Waypoint)
```

Comment:
Why do we need this delegate method addition when we already have this? https://mapbox.github.io/mapbox-navigation-ios/navigation/0.7.0/Protocols/NavigationViewControllerDelegate.html#/s:FP16MapboxNavigation32NavigationViewControllerDelegate24navigationViewControllerFTCS_24NavigationViewController11didArriveAtC16MapboxDirections8Waypoint_T_

Reply:
Copying here for posterity: the idea here is that it establishes the `RouteController` as the authority on location updates, and enforces order-of-operations, which would be very helpful to an implementer. You cannot ensure order-of-operations of a notification, but you can easily override a delegate method in a subclass. The idea here is that if someone wanted to do something custom with `NavigationViewController` where they wanted to do some custom logic before firing off the `navigationViewController(_:didArriveAt:)`, they could.